mlflow/types/schema.py

```diff
@@ -540,7 +540,13 @@ def enforce_param_datatype(cls, name, value, dtype: DataType):
 
         if dtype == DataType.datetime:
             try:
-                return np.datetime64(value).item()
+                datetime_value = np.datetime64(value).item()
+                if isinstance(datetime_value, int):
+                    raise MlflowException.invalid_parameter_value(
+                        f"Invalid value for param {name}, it should "
+                        f"be convertible to datetime.date/datetime, got {value}"
+                    )
```

Comment:
I think we don't need this because `datetime_value = np.datetime64(value).item()` already converts the value ?

Reply:
The strange thing is that `np.datetime64('20230701').item()` is actually an integer 😅